### PR TITLE
Prioritize IPv4 addresses in the enrollment token

### PR DESCRIPTION
**Pull request by jkakavas**
The enrollment token contains a list of addreses where the node is
listening for requests in the HTTP layer. This change makes it so
we order that list before serializing it in the token prioritizing
IPv4 addresses bs IPv6 addresses. The reasoning is that
- it is usually more probable for IPv4 addresses to be reachable
so in the case where the consumer of the token needs to iterate
through available addresses, this might lead to getting a response
with fewer attempts
- When persisted to configuration files and/or shown in UIs, IPv4
addresses look more familiar (less surprising) to users

### CODE REVIEW

**Comment by jkakavas** [comment]
cc @azasypkin

**Review by albertzaharovits — APPROVED** [review]

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/EnrollmentTokenGenerator.java [hunk 1]
```diff
@@ -199,14 +201,38 @@ protected String getCaFingerprint() throws Exception {
     }
 
     static List<String> getFilteredAddresses(List<String> addresses) throws Exception {
-        List<String> filtered_addresses = new ArrayList<>();
+        List<String> filteredAddresses = new ArrayList<>();
         for (String bound_address : addresses){
```
Comment (BigPandaToo) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/EnrollmentTokenGenerator.java:205]: Just as we are changing the names according to the convention, can we also rename it to `boundAddress`?

**Review by BigPandaToo — APPROVED** [review]
I added one nit comment; LGTM otherwise

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/part-2-fips

**Pull request merged**